Tests/RegexTests/MatchTests.swift

```diff
@@ -1575,27 +1577,57 @@ extension RegexTests {
       (input: "hezllo", match: nil),
       (input: "helloz", match: nil))
 
+    // MARK: Lookbehinds
     firstMatchTest(
-      #"(?<=USD)\d+"#, input: "Price: USD100", match: "100", xfail: true)
+      #"(?<=USD)\d+"#, input: "Price: USD100", match: "100")
     firstMatchTest(
-      #"(*plb:USD)\d+"#, input: "Price: USD100", match: "100", xfail: true)
+      #"(*plb:USD)\d+"#, input: "Price: USD100", match: "100")
     firstMatchTest(
       #"(*positive_lookbehind:USD)\d+"#,
-      input: "Price: USD100", match: "100", xfail: true)
-    // engines generally enforce that lookbehinds are fixed width
+      input: "Price: USD100", match: "100")
+
+    // TODO: Why is a match not found when unoptimized?
     firstMatchTest(
-      #"\d{3}(?<=USD\d{3})"#, input: "Price: USD100", match: "100", xfail: true)
+      #"\d{3}(?<=USD\d{3})"#, input: "Price: USD100", match: "100", validateOptimizations: false)
 
     firstMatchTest(
-      #"(?<!USD)\d+"#, input: "Price: JYP100", match: "100", xfail: true)
+      #"(?<!USD)\d+"#, input: "Price: JYP100", match: "100")
     firstMatchTest(
-      #"(*nlb:USD)\d+"#, input: "Price: JYP100", match: "100", xfail: true)
+      #"(*nlb:USD)\d+"#, input: "Price: JYP100", match: "100")
     firstMatchTest(
       #"(*negative_lookbehind:USD)\d+"#,
-      input: "Price: JYP100", match: "100", xfail: true)
-    // engines generally enforce that lookbehinds are fixed width
+      input: "Price: JYP100", match: "100")
+
+    firstMatchTest(
+      #"\d{3}(?<!USD\d{3})"#, input: "Price: JYP100", match: "100")
+
+    firstMatchTest(#"(?<=abc)def"#, input: "abcdefg", match: "def", validateOptimizations: false)
+    firstMatchTests(
+      #"(?<=az|b|c)def"#,
+      ("azdefg", "def"),
+      ("bdefg", "def"),
+      ("cdefg", "def"),
+      ("123defg", nil),
+      validateOptimizations: false
+    )
+
+    // FIXME: quickMatch and thoroughMatch have different results
     firstMatchTest(
-      #"\d{3}(?<!USD\d{3})"#, input: "Price: JYP100", match: "100", xfail: true)
+      #"(?<=\d{1,3}-.{1,3}-\d{1,3})suffix"#,
+      input: "123-_+/-789suffix",
+      match: "suffix",
+      validateOptimizations: false
+    )
+
+    firstMatchTests(
+      #"(?<=^\d{1,3})abc"#,
+      ("123abc", "abc"),
+      ("12abc", "abc"),
+      ("1abc", "abc"),
+      ("1234abc", nil), // FIXME: Shouldn't match but does because `^` assertions are broken
+      ("z123abc", nil), // FIXME: Same as above
+      validateOptimizations: false
+    )
```

Comment:
Let's see if we can have nested lookarounds (like a positive inside a negative inside a positive, etc). You can cross reference other engine's behavior such as Javascript and .NET (e.g., via https://regex101.com).